bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/rewriter/gemini/QueryVisitorRewriter.java

```diff
@@ -0,0 +1,91 @@
+package com.google.zetasql.toolkit.antipattern.rewriter.gemini;
+
+import com.google.zetasql.SqlException;
+import com.google.zetasql.toolkit.antipattern.AntiPatternVisitor;
+import com.google.zetasql.toolkit.antipattern.cmd.InputQuery;
+import com.google.zetasql.toolkit.antipattern.rewriter.prompt.PromptYamlReader;
+import com.google.zetasql.toolkit.antipattern.util.AntiPatternHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class QueryVisitorRewriter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryVisitorRewriter.class);
+    private final PromptYamlReader promptYamlReader;
+    private final AntiPatternHelper antiPatternHelper;
+
+    public QueryVisitorRewriter(AntiPatternHelper antiPatternHelper) throws IOException {
+        this.promptYamlReader = new PromptYamlReader();
+        this.antiPatternHelper = antiPatternHelper;
+    }
+
+    public String rewriteSQL(String inputQuery,
```

Comment:
imo the retry capability in the function is not immediately intuitive to a first time reader. Could we please add some comments to provide more context?

Reply:
Let me check if it makes more sense now, with the change of name of the variable

Reply:
can we just add a comment somewhere saying that retries will be used:

1. In case the antipattern is still there
2. In case the query is syntactically invalid

Reply:
Done